src/libmongoc/src/mongoc/mongoc-cyrus.c

```diff
@@ -149,19 +149,19 @@ _mongoc_cyrus_verifyfile_cb (void *context, const char *file, sasl_verify_type_t
    TRACE ("Attempting to load file: `%s`. Type is %s\n", file, sasl_verify_type_to_str (type));
 
 #ifdef _WIN32
-   // On Windows, Cyrus-SASL hard-codes the plugin path.
-   // Only permit loading plug-in from user configured path to prevent unintentional library loading.
+   // On Windows, Cyrus SASL hard-codes the plugin path.
+   // Only permit loading plugin from user configured path to prevent unintentional library loading.
    if (type == SASL_VRFY_PLUGIN) {
       const char *path_prefix = MONGOC_CYRUS_PLUGIN_PATH_PREFIX;
-      bool has_valid_prefix = (0 != strlen (path_prefix) && file == strstr (file, path_prefix));
+      bool has_valid_prefix = (path_prefix && file == strstr (file, path_prefix));
       // Check if `file` has necessary prefix.
       if (has_valid_prefix) {
          return SASL_OK;
       }
-      MONGOC_WARNING ("Not loading Cyrus-SASL plugin: %s. If plugin is needed, set CMake option "
-                      "`CYRUS_PLUGIN_PATH_PREFIX (currently '%s')` to a non-empty string contain the path prefix.",
+      MONGOC_WARNING ("Refusing to load Cyrus SASL plugin at: '%s'. If needed, set CYRUS_PLUGIN_PATH_PREFIX (currently "
+                      "'%s') to the absolute path prefix during build configuration.",
```

Comment:
```suggestion
                      "'%s') to the absolute path prefix of the plugin during build configuration of the C Driver.",
```
Avoid confusing build configuration of the user's project vs. build configuration of the C Driver.